[RBAC]: Document how they work on the FE in `contributors`

joshuaellis opened the pull request:
<!--
Hello 👋 Thank you for submitting a pull request.

To help us merge your PR, make sure to follow the instructions below:

- Create or update the tests
- Create or update the documentation at https://github.com/strapi/documentation
- Refer to the issue you are closing in the PR description: Fix #issue
- Specify if the PR is ready to be merged or work in progress (by opening a draft PR)

Please ensure you read the Contributing Guide: https://github.com/strapi/strapi/blob/main/CONTRIBUTING.md
-->

### What does it do?

* Documents RBAC and how it's used on the FE

### Why is it needed?

* Because no dev actually remembers 💀
* Helps us see how it looks from a high level and how we can simplify it

### Related issue(s)/PR(s)

* resolves CONTENT-994

joshuaellis commented:
First page done 👍🏼

ronronscelestes commented on docs/docs/core/permissions/frontend/fetching-permissions.mdx:
```diff
@@ -0,0 +1,96 @@
+- Signing into the app
+- Not Found
+- Collecting information from a first time user
+- The actual app
```
I think I would be more specific about the actual app being the Authenticated app, wdyt? 👀

ronronscelestes commented on docs/docs/core/permissions/frontend/fetching-permissions.mdx:
```diff
@@ -0,0 +1,96 @@
+- Collecting information from a first time user
+- The actual app
+
+Permissions, along with other vital application information is fetched in in the AuthenticatedApp component, located
```
small UI detail: `AuthenticatedApp` instead of AuthenticatedApp?

ronronscelestes commented on docs/docs/core/permissions/frontend/fetching-permissions.mdx:
```diff
@@ -0,0 +1,96 @@
+
+## What do we do with the data?
+
+The data is then passed to the `RBACProvider` comoponent which passes it to a public facing [React Context](https://beta.reactjs.org/learn/passing-data-deeply-with-context)
```
small typo: comoponent vs component

ronronscelestes commented on docs/docs/core/permissions/frontend/fetching-permissions.mdx:
```diff
@@ -0,0 +1,96 @@
+  alt="diagram of how permissions get from the server to the application"
+/>
+
+Inside the `RBACProviderContext` we also pass the ability to refetch the permissions. These two inconjunction with one another are accessed
```
wdyt of specifying the name of what is passed? (e.g. `allPermissions` & `refetchPermissions`)

joshuaellis replied:
Added a link to the API page

ronronscelestes reviewed (commented):
so nice to have a first overview of this part of the code - thank you!
small comments on fetching permissions 🎣

joshuaellis commented:
@ronronscelestes i've added the API reference page and addressed those comments 👍🏼 

After monday's meeting we should be good to write the `how-they-work` page and get others to review it.

joshuaellis commented on docs/docs/core/permissions/how-they-work.mdx:
````diff
@@ -0,0 +1,20 @@
+
+```ts
+const permissions = {
+  create: [{ action: 'plugin::content-manager.explorer.create', subject: null }],
````
@petersg83

Would you be able to give a small paragraph how the `string` works? and what a `subject` is please? From my understanding so far, `subject` only applies to `content-types` but that's probably wrong?

petersg83 replied:
on it

petersg83 replied:
Done, what do you think?

joshuaellis replied:
They are really good, thank you and well done 🙇🏼‍♀️

Marc-Roig commented on docs/docs/core/permissions/how-they-work.mdx:
```diff
@@ -0,0 +1,150 @@
+
+| key                           | description                                                                                                                                       | type   | required | Default value | example                    |
+| ----------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- | ------ | -------- | ------------- | -------------------------- |
+| **uid**                       | An id that has to be unique within the plugin.                                                                                                    | string | required | -             | `'audit-logs.read'`        |
```
On what do we base this uid?

Something like <nameoffeature>.<action> ?

petersg83 replied:
It depends, it can be the name of the entity. It's nice to follow how it was done before but it's not strictly consistent. But ultimately the only thing that matter is that it is unique.
You can find a list of existing ids here: https://docs.google.com/spreadsheets/d/13kIEKshstD9WJqWBctxb8TeAIHDZCw5ttg7vgzUAsBU/edit#gid=0

joshuaellis commented:
Hey devs (not gonna tag you all) me and @ronronscelestes with the help of @petersg83 documented permissions on the FE.

It'd be great for you all to read, see if it makes sense to you all and if you have any questions, comment so we can address them as there might be more info we don't realise is needed! :)

simotae14 commented on docs/docs/core/permissions/frontend/using-permissions.mdx:
```diff
@@ -0,0 +1,283 @@
+global permissions object generated from the [`useRBACProvider`](#userbacprovider) hook which can either be passed or will be accessed internally.
+
+Because it's fetching, we also provide a `isLoading` state which can be used to show a loading state while the permissions are being fetched.
+If the hook is unmouted befor the fetch request completes, the request will be cancelled via an [`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController).
```
typo
befor => before

simotae14 commented on docs/docs/core/permissions/intro.mdx:
```diff
@@ -0,0 +1,27 @@
+This is not about the `user-permissions` plugin which is API orientated, this is about roles based access control commonly known as RBAC.
+:::
+
+This section is an overview of all the features realted to the Permissions core plugin:
```
typo
realted -> related

simotae14 reviewed (dismissed):
I found two typos, overall I think it is really good

ronronscelestes commented on docs/docs/core/permissions/how-they-work.mdx:
```diff
@@ -0,0 +1,150 @@
+| **category**                  | Name of the category. It will define in which category the action will appear. _Only for the plugins and settings section_.                       | string | -        | -             | `'audit logs'`             |
+| **subCategory**               | Name of the subcategory. It will define in which subcategory or the category the action will appear. _Only for the plugins and settings section_. | string | -        | -             | `'options'`                |
+| **displayName**               | Human name of the action.                                                                                                                         | string | required | -             | `'Read'`                   |
+| **subjects**                  | List of subjects the action can be applied to. _Only for the contentTyes section_.                                                                | array  | -        | -             | `['api::article.article']` |
```
```suggestion
| **subjects**                  | List of subjects the action can be applied to. _Only for the contentTypes section_.                                                                | array  | -        | -             | `['api::article.article']` |
```

same for `options.applyToProperties` 👀

ronronscelestes reviewed (commented):
found a small typo - other than that looks perfect 👌

simotae14 reviewed (approved):
lgtm
